### PR TITLE
feat(alpha): monitoring task listing and model change

### PR DESCRIPTION
**Pull request by VerstraeteBert**
- Listing endpoint: only allow querying by TimeseriesId, not by TimeseriesExternalId
- Create endpoint: embed ModelExternalId and parameters into Model field

### CODE REVIEW

File: packages/alpha/src/types.ts [hunk 1]
```diff
@@ -17,47 +17,64 @@ import {
...
+  constructor(
+    timeseriesExternalId: CogniteExternalId,
+    threshold: number,
+    granularity?: string
+  ) {
+    this.timeseriesExternalId = timeseriesExternalId;
+    this.granularity = granularity;
+    this.threshold = threshold;
+  }
+}
```
Comment (polomani) [inline packages/alpha/src/types.ts:31]: we don't do classes here, this is overly complex for the SDK :)
```
export interface MonitoringTaskThresholdModelCreate {
  externalId: "threshold";
  timeseriesExternalId: CogniteExternalId;
  threshold: number;
  granularity?: string;
```
should be enough

File: packages/alpha/src/types.ts [hunk 2]
```diff
@@ -17,47 +17,64 @@ import {
 
 export * from '@cognite/sdk';
 
-export interface MonitoringTaskParametersCreate {
-  timeseriesExternalId: CogniteExternalId;
-  granularity?: string;
-  threshold: number;
+enum ModelExternalId {
+  Threshold = 'threshold',
+}
```
Comment (polomani) [inline packages/alpha/src/types.ts]: please do not use enums they're quite painful for the consumer.

`paramName: "string_val1" | "string_val1"` is usually fine

File: packages/alpha/src/types.ts (hunk 1 above)
Comment (VerstraeteBert) [reply packages/alpha/src/types.ts:31]: Tried this earlier. Interfaces don't seem to play well with literals, when constructing this and using it usually complains that (similar to) ` externalId: type "threshold" expected, got type string. `. Can mitigate this by not duck typing anymore and explicitly casting the object MonitoringTaskThresholdModelCreate (see new commit), which is not ideal either.

Comment (VerstraeteBert) [reply packages/alpha/src/types.ts:31]: Now users both have to declare explicitely which Model they're using in the cast and redundantly in the object iself with the externalId. By using the class and hidden enum this is not necessary.

File: packages/alpha/src/types.ts (hunk 2 above)
Comment (VerstraeteBert) [reply packages/alpha/src/types.ts]: Enums were hidden for users, only used internally.

File: packages/alpha/src/types.ts (hunk 1 above)
Comment (polomani) [reply packages/alpha/src/types.ts:31]: classes give different sorts of headaches, trust me on this one. js SDK has a long history of going from interfaces to classes and then back to interfaces.

If you wish to improve the developer experience you can define a type + const.
Example: 

```
export type ContextJobStatus = 'QUEUED' | 'RUNNING' | 'COMPLETED' | 'FAILED';

export const ContextJobStatus = {
  QUEUED: 'QUEUED' as ContextJobStatus,
  RUNNING: 'RUNNING' as ContextJobStatus,
  COMPLETED: 'COMPLETED' as ContextJobStatus,
  FAILED: 'FAILED' as ContextJobStatus,
};
```

File: packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts [hunk 3]
```diff
@@ -21,14 +21,20 @@ describe('monitoring tasks api', () => {
   const monitoringTaskExternalId = `test_mt_${ts}`;
   const channelExternalId = `test_channel_mt_${ts}`;
   const sessionsApi = `/api/v1/projects/${TEST_PROJECT}/sessions`;
-  const testMtParams = {
+  const testMtModel = {
```
Comment (polomani) [inline packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts]:
```suggestion
  const testMtModel: MonitoringTaskThresholdModelCreate
```
to get a proper type inference, as you declare this outside of the request

Comment (polomani) [reply packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts]: similar to `model: testMtModel as MonitoringTaskThresholdModelCreate` but better

**Comment by polomani** [comment]
`yarn lint:fix` + remove the comments

File: packages/alpha/src/types.ts [hunk 4]
```diff
@@ -17,47 +17,53 @@ import {
 
 export * from '@cognite/sdk';
 
-export interface MonitoringTaskParametersCreate {
+export type ModelExternalId = 'threshold';
+
+export const ModelExternalId = {
```
Comment (polomani) [inline packages/alpha/src/types.ts]:
```suggestion
export const MonitoringTaskModelExternalId = {
```
prepend API name for users to get a nice autocomplete

File: packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts [hunk 5]
```diff
@@ -1,6 +1,10 @@
 // Copyright 2022 Cognite AS
 
-import { Channel } from 'alpha/src/types';
+import {
+  Channel,
+  ModelExternalId,
+  MonitoringTaskThresholdModelCreate,
+} from 'alpha/src/types';
```
Comment (polomani) [inline packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts]:
```suggestion
} from '../../types';
```

Comment (polomani) [reply packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts]: this is the import issue you have

File: packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts [hunk 6]
```diff
@@ -21,14 +25,20 @@ describe('monitoring tasks api', () => {
   const monitoringTaskExternalId = `test_mt_${ts}`;
   const channelExternalId = `test_channel_mt_${ts}`;
   const sessionsApi = `/api/v1/projects/${TEST_PROJECT}/sessions`;
-  const testMtParams = {
+  const testMtModel: MonitoringTaskThresholdModelCreate = {
+    externalId: ModelExternalId.THRESHOLD,
+    timeseriesExternalId: 'test-functions',
```
Comment (polomani) [inline packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts]:
```suggestion
    timeseriesExternalId: 'test_functions',
```

**Review by polomani — APPROVED** [review]

File: packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts [hunk 7]
```diff
@@ -1,6 +1,10 @@
 // Copyright 2022 Cognite AS
 
-import { Channel } from 'alpha/src/types';
+import {
+  Channel,
+  ModelExternalId,
```
Comment (polomani) [inline packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts]:
```suggestion
  MonitoringTaskModelExternalId,
```

File: packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts [hunk 8]
```diff
@@ -21,14 +25,20 @@ describe('monitoring tasks api', () => {
   const monitoringTaskExternalId = `test_mt_${ts}`;
   const channelExternalId = `test_channel_mt_${ts}`;
   const sessionsApi = `/api/v1/projects/${TEST_PROJECT}/sessions`;
-  const testMtParams = {
-    threshold: 50.1,
+  const testMtModel: MonitoringTaskThresholdModelCreate = {
+    externalId: ModelExternalId.THRESHOLD,
```
Comment (polomani) [inline packages/alpha/src/__tests__/api/monitoringTasksApi.int.spec.ts]:
```suggestion
    externalId: MonitoringTaskModelExternalId.THRESHOLD,
```

**Review by hamdielhamdi-cognite — APPROVED** [review]

**Review by sou1est — APPROVED** [review]

**Pull request merged**